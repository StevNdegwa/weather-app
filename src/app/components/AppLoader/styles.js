import { styled, Box } from "@mui/material";

export const AppLoaderContent = styled(Box)`
width: 100%;
height: 100%;
display: flex;
justify-content: center;
align-items: center;
font-size:1.3rem;
font-weight:800;
color: darkgray;
background-color:transparent;
`;

export const WeatherIconContainer = styled(Box)`
position: absolute;
top:0;
right:0;
margin:10px;
`;