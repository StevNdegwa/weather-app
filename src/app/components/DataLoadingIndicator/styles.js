import { Box, styled } from "@mui/material";

export const LoadingIndicatorWrapper = styled(Box)`
position: absolute;
top: 0;
left: 0;
z-index: var(--depth-overlay);
width: 100%;
height: 100%;
display: flex;
justify-content: center;
align-items: center;
background-color: hsla(0, 0%, 100%, 0.2);
backdrop-filter: blur(3px);
`;
