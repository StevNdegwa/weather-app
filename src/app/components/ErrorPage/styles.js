import { styled, Box } from "@mui/material";

export const ErrorPageWrapper = styled(Box)`
width: 100%;
height: 100%;
display: flex;
justify-content: center;
align-items: center;
`;

export const ErrorPageContent = styled(Box)`
text-align:center;
font-size: var(--font-size-md);
`;
