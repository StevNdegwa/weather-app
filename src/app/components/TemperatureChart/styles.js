import { Stack, styled } from "@mui/material";

export const ChartWrapper = styled(Stack)`
max-width: 700px;
height: 400px;
background-color: white;
position: relative;
border-radius: var(--border-radius-md);
margin: 5rem auto;
overflow: auto;
& > div{
  width: 700px;
  height:100%;
  padding: 0.5rem;
}
@media only screen and (max-width: 800px){
  margin: 0 auto;
  border-radius: 0;
}
& svg{
  font-size: 0.8rem;
  & path{
    fill: #90CBEC;
    cursor: pointer;
  }
}
`;