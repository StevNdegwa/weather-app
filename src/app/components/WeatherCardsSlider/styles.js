import { styled, Slider } from "@mui/material";
import { Slide } from "react-slideshow-image";

export const WeatheCardsSliderWrapper = styled(Slide)`
width: 95%;
background-color: none;
overflow: hidden;
border-radius: var(--border-radius-lg);
& button:disabled{
  opacity: 0;
}
`;

export const IndexSlider = styled(Slider)`
pointer-events: none;
& .MuiSlider-track {
  border: none;
  background-color: #90CBEC !important;
}
& .MuiSlider-rail {
  border: none;
  background-color: #DAE3E8 !important;
}
& .MuiSlider-thumb{
  display: none;
}
`;
