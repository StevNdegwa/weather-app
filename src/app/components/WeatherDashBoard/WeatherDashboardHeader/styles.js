import { styled, Box } from "@mui/material";

export const HeaderWrapper = styled(Box)`
width: 100%;
height: 60px;
display: flex;
justify-content: space-between;
align-items: center;
padding:0 1rem;
`;
