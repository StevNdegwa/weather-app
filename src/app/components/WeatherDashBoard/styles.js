import { Box, Stack, styled } from "@mui/material";

export const WeatherDashBoardWrapper = styled(Box)`
width: 100%;
height: 100%;
display: flex;
justify-content:center;
align-items: center;
`;

export const WeatherDashBoardContent = styled(Stack)`
max-width:800px;
width: 100%;
max-height: 600px;
height: 100%;
`;
