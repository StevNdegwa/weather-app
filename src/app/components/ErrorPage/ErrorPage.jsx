import { Button, Typography } from "@mui/material";
import PropTypes from "prop-types";
import { useCallback } from "react";
import { GrRefresh } from "react-icons/gr";
import { ErrorPageWrapper, ErrorPageContent } from "./styles";

export default function ErrorPage({ errorMessage }) {
  const reloadApp = useCallback(() => {
    window.location.reload();
  }, []);

  return (
    <ErrorPageWrapper>
      <ErrorPageContent>
        <Typography>{errorMessage}</Typography>
        <Button startIcon={<GrRefresh />} onClick={reloadApp}>
          Refresh application
        </Button>
      </ErrorPageContent>
    </ErrorPageWrapper>
  );
}

ErrorPage.propTypes = {
  errorMessage: PropTypes.string.isRequired,
};
