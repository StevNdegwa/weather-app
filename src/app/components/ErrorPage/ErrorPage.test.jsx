import { mount } from "enzyme";
import { act } from "@testing-library/react";
import { Button } from "@mui/material";
import { ErrorPageContent } from "./styles";
import ErrorPage from "./ErrorPage";

describe("test <ErrorPage/> component", () => {
  let wrapper,
    errorMessage = "A bad error",
    windowSpy;

  beforeEach(() => {
    windowSpy = jest.spyOn(window, "window", "get");
    wrapper = mount(<ErrorPage errorMessage={errorMessage} />);
  });

  afterEach(() => {
    windowSpy.mockRestore();
  });

  test("snapshot testing", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should render error message", () => {
    expect(wrapper.find(ErrorPageContent).text()).toEqual(
      expect.stringContaining(errorMessage)
    );
  });

  test("should reload application", async () => {
    let reload = jest.fn();

    windowSpy.mockImplementation(() => ({
      location: {
        reload,
      },
    }));

    await act(async () => {
      wrapper.find(Button).prop("onClick")();
    });

    expect(reload).toHaveBeenCalled();
  });
});
