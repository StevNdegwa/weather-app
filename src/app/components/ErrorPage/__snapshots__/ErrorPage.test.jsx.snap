// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <ErrorPage/> component snapshot testing 1`] = `"<div class=\\"MuiBox-root css-mpb51q\\"><div class=\\"MuiBox-root css-12ss6jc\\"><p class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\">A bad error</p><button class=\\"MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" d=\\"M20,8 C18.5974037,5.04031171 15.536972,3 12,3 C7.02943725,3 3,7.02943725 3,12 C3,16.9705627 7.02943725,21 12,21 L12,21 C16.9705627,21 21,16.9705627 21,12 M21,3 L21,9 L15,9\\"></path></svg></span>Refresh application<span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div>"`;
